presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoPreparedStatement.java

```diff
@@ -33,128 +36,163 @@
     }
 
     @Override
     public void setByte(int parameterIndex, byte x)
             throws SQLException
     {
-        throw new NotImplementedException("PreparedStatement", "setByte");
+        checkOpen();
+        setParameter(parameterIndex, formatCast(formatLongLiteral(x), Types.TINYINT));
     }
 
     @Override
     public void setShort(int parameterIndex, short x)
             throws SQLException
     {
-        throw new NotImplementedException("PreparedStatement", "setShort");
+        checkOpen();
+        setParameter(parameterIndex, formatCast(formatLongLiteral(x), Types.SMALLINT));
     }
 
     @Override
     public void setInt(int parameterIndex, int x)
             throws SQLException
     {
-        throw new NotImplementedException("PreparedStatement", "setInt");
+        checkOpen();
+        setParameter(parameterIndex, formatCast(formatLongLiteral(x), Types.INTEGER));
     }
 
     @Override
     public void setLong(int parameterIndex, long x)
             throws SQLException
     {
-        throw new NotImplementedException("PreparedStatement", "setLong");
+        checkOpen();
+        setParameter(parameterIndex, formatCast(formatLongLiteral(x), Types.BIGINT));
     }
 
     @Override
     public void setFloat(int parameterIndex, float x)
             throws SQLException
     {
-        throw new NotImplementedException("PreparedStatement", "setFloat");
+        checkOpen();
+        setParameter(parameterIndex, formatCast(formatDoubleLiteral(x), Types.REAL));
     }
 
     @Override
     public void setDouble(int parameterIndex, double x)
             throws SQLException
     {
-        throw new NotImplementedException("PreparedStatement", "setDouble");
+        checkOpen();
+        setParameter(parameterIndex, formatCast(formatDoubleLiteral(x), Types.DOUBLE));
     }
 
     @Override
     public void setBigDecimal(int parameterIndex, BigDecimal x)
             throws SQLException
     {
-        throw new NotImplementedException("PreparedStatement", "setBigDecimal");
+        checkOpen();
+        setParameter(parameterIndex, formatDecimalLiteral(x));
     }
 
     @Override
     public void setString(int parameterIndex, String x)
             throws SQLException
     {
-        throw new NotImplementedException("PreparedStatement", "setString");
+        checkOpen();
+        setParameter(parameterIndex, formatStringLiteral(x));
     }
 
     @Override
     public void setBytes(int parameterIndex, byte[] x)
             throws SQLException
     {
-        throw new NotImplementedException("PreparedStatement", "setBytes");
+        checkOpen();
+        setParameter(parameterIndex, formatBinaryLiteral(x));
     }
 
     @Override
     public void setDate(int parameterIndex, Date x)
             throws SQLException
     {
-        throw new NotImplementedException("PreparedStatement", "setDate");
+        checkOpen();
+        setParameter(parameterIndex, formatDateLiteral(DATE_FORMATTER.print(x.getTime())));
```

Comment:
We're using the Joda-Time formatters that are used to parse values in `PrestoResultSet`. It doesn't support JDK 8 temporal types.

Can you elaborate on the suggestion? Is the current code incorrect?

Reply:
It's _correct (obviously:)_, but I would prefer it to be _obviously correct_.
I.e. you want to render `DATE 'yyyy-MM-dd'` literal from a `Date` value. Going through `x.getTime()` causes a "CPU spike" in my head since I need to think twice to know this is correct.
However, this code could be simplified down to 
```
java.time.format. DateTimeFormatter.ofPattern("'DATE '''yyyy-MM-dd''").format(date.toLocalDate())
```
(no `long millis` involved here)